crates/client/mempool/src/block_production.rs

```diff
@@ -308,7 +319,7 @@ impl BlockProductionTask {
         Ok(())
```

Comment:
so ignoring the first tick means 

 BLOCK N BLOCK N+1
|.......|. -> this one is ignored or just the | so the time block is finalized?

For this to work we have to make sure both clocks are synchronized tho

Reply:
Also `continue_block_production` is probably a better name

Reply:
> For this to work we have to make sure both clocks are synchronized tho

I think I need to add a a reset to the pending block interval yes, something like
```rs
instant = interval_block_time.tick() => {
    if let Err(err) = self.produce_block().await {
        log::error!("Block production task has errored: {err:#}");
    }
    // this sets the next deadline to ensure the pending block tick and block time always match up
    interval_pending_block_update.reset_at(instant + interval_pending_block_update.period());
},
```
i'd like to have tests around that part that use tokio::time::advance to skip block times, and see how well this works because i'm almost sure i missed other issues

> Also continue_block_production is probably a better name

there is already a function named "continue_block" :(

Reply:
we also talked about removing tokio entirely in the block production because of DoS issues, right now i'm leaning towards making another tokio thread pool with higher thread priorities instead but we may switch all of this to hand written non async timers
I think just having a dedicated tokio pool is fine, I don't see why this would have any impact on perf vs a dedicated thread even if we don't use it that much